internal/handler/Handler.go

```diff
@@ -0,0 +1,281 @@
+	jsonBytes, err := json.Marshal(employees)
+	if err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "Error: bad request", http.StatusBadRequest)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) getPosition(w http.ResponseWriter, r *http.Request) {
+	matches := positionRe.FindStringSubmatch(r.URL.Path)
+	if len(matches) < 2 {
+		http.NotFound(w, r)
+		return
+	}
+	p, err := h.service.GetPosition(matches[1])
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		_, err := w.Write([]byte("Position not found"))
+		if err != nil {
+			http.Error(w, "Error: bad request", http.StatusBadRequest)
+		}
+		return
+	}
+	jsonBytes, err := json.Marshal(p)
+	if err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "Error: bad request", http.StatusBadRequest)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) getEmployee(w http.ResponseWriter, r *http.Request) {
+	matches := employeeRe.FindStringSubmatch(r.URL.Path)
+	if len(matches) < 2 {
+		http.NotFound(w, r)
+		return
+	}
+	e, err := h.service.GetEmployee(matches[1])
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		_, err := w.Write([]byte("user not found"))
+		if err != nil {
+			http.Error(w, "Error: bad request", http.StatusBadRequest)
+		}
+		return
+	}
+	jsonBytes, err := json.Marshal(e)
+	if err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "Error: bad request", http.StatusBadRequest)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) createPosition(w http.ResponseWriter, r *http.Request) {
+	var p internal.Position
+	if err := json.NewDecoder(r.Body).Decode(&p); err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	err:=h.service.CreatePosition(p)
+	if err != nil{
+		http.Error(w,err.Error(),http.StatusNotFound)
+	}
+	jsonBytes, err := json.Marshal(p)
+	if err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "Error: bad request", http.StatusBadRequest)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) createEmployee(w http.ResponseWriter, r *http.Request) {
+	var e internal.Employee
+	if err := json.NewDecoder(r.Body).Decode(&e); err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	err:=h.service.CreateEmployee(e)
+	if err != nil{
+		http.Error(w,err.Error(),http.StatusNotFound)
+	}
+	jsonBytes, err := json.Marshal(e)
+	if err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "Error: bad request", http.StatusBadRequest)
```

Comment:
StatusInternalServerError is better here

Reply:
+